src/lib.rs

```diff
@@ -287,27 +280,25 @@ mod tests {
             )
             .unwrap();
 
-            let mut output = vec![0u8; 64];
+            let mut output = [0u8; 64];
 
-            let res = bn128_mul(&input[..], &mut BytesRef::Fixed(&mut output[..]));
+            let res = bn128_mul(&input[..], &mut output);
             assert!(res.is_err(), "There should be built-in error here");
         }
     }
 
     fn pairing_empty_test(expected: Vec<u8>) {
         let mut empty = [0u8; 0];
-        let input = BytesRef::Fixed(&mut empty);
-
-        let mut output = vec![0u8; expected.len()];
+        let input = &mut empty;
```

Comment:
Input doesn't need to be mutable (but in the old example the field was reused).

Reply:
thats true